src/vs/workbench/contrib/notebook/browser/services/notebookExecutionStateServiceImpl.ts

```diff
@@ -142,11 +147,25 @@ export class NotebookExecutionStateService extends Disposable implements INotebo
 	private _createNotebookCellExecution(notebook: NotebookTextModel, cellHandle: number): CellExecution {
 		const notebookUri = notebook.uri;
 		const exe: CellExecution = this._instantiationService.createInstance(CellExecution, cellHandle, notebook);
+		const cellUri = CellUri.generate(notebookUri, cellHandle);
+		const audioCuesEnabled = this._configurationService.getValue<boolean>(NotebookSetting.audioCuesEnabled);
+
+		let onDidCompleteCallback: (lastRunSuccess: boolean | undefined) => void;
+		if (audioCuesEnabled) {
+			const audioCuesCallback = this._getAudioCuesOnDidCompleteCallback(cellUri);
+			onDidCompleteCallback = (lastRunSuccess) => {
+				this._onCellExecutionDidComplete(notebookUri, cellHandle, exe, lastRunSuccess);
+				audioCuesCallback(lastRunSuccess);
+			};
+		} else {
+			onDidCompleteCallback = (lastRunSuccess) => this._onCellExecutionDidComplete(notebookUri, cellHandle, exe, lastRunSuccess);
+		}
+
 		const disposable = combinedDisposable(
 			exe.onDidUpdate(() => this._onCellExecutionDidChange(notebookUri, cellHandle, exe)),
-			exe.onDidComplete(lastRunSuccess => this._onCellExecutionDidComplete(notebookUri, cellHandle, exe, lastRunSuccess)));
-		this._cellListeners.set(CellUri.generate(notebookUri, cellHandle), disposable);
-
+			exe.onDidComplete(onDidCompleteCallback));
+		this._cellListeners.set(cellUri, disposable);
+		this._lastScheduledCellUri = cellUri;
```

Comment:
Not crazy about this approach- this service is global where we should track this per-notebook. And this tracks the last cell to start, and as Megan points out, in some notebooks cells can run in parallel and thus finish out of order. I think a better way to do this would be when a cell finishes running, check whether there are any other executing cells in the same notebook, if not, play the sound.

Reply:
Reverted to the old style + implemented the behaviour in case of multiple cell executed